libraries/Microsoft.Bot.Connector/Authentication/ParameterizedBotFrameworkAuthentication.cs

```diff
@@ -104,22 +104,28 @@ private async Task<ClaimsIdentity> JwtTokenValidation_AuthenticateRequestAsync(A
             if (string.IsNullOrWhiteSpace(authHeader))
             {
                 var isAuthDisabled = await credentialFactory.IsAuthenticationDisabledAsync(cancellationToken).ConfigureAwait(false);
-                if (isAuthDisabled)
+                if (!isAuthDisabled)
                 {
-                    // In the scenario where Auth is disabled, we still want to have the
-                    // IsAuthenticated flag set in the ClaimsIdentity. To do this requires
-                    // adding in an empty claim.
-                    return new ClaimsIdentity(new List<Claim>(), "anonymous");
+                    // No Auth Header. Auth is required. Request is not authorized.
+                    throw new UnauthorizedAccessException();
                 }
 
-                // No Auth Header. Auth is required. Request is not authorized.
-                throw new UnauthorizedAccessException();
+                // Check if the activity is for a skill call and is coming from the Emulator.
+                if (activity.ChannelId == Channels.Emulator && activity.Recipient?.Role == RoleTypes.Skill)
+                {
+                    // Return an anonymous claim with an anonymous skill AppId
+                    return new ClaimsIdentity(new List<Claim> { new Claim(AuthenticationConstants.AppIdClaim, AuthenticationConstants.AnonymousSkillAppId) }, AuthenticationConstants.AnonymousAuthType);
```

Comment:
We're doing this in 3 places now. If we change one, we are always going to change the other. Are we sure we don't want to abstract this?

Reply:
If you want to validate this solution in composer and all that is fine, but eventually once we are happy with this solution, we need to make this code centralized. Otherwise hard to prevent future maintainers to change only one of the places and break some scenarios.